deeplab/model.py

```diff
@@ -0,0 +1,262 @@
+
+    # specify which layers to extract activations from
+    return_layers = {
+        'layer1': 'layer1',
+        'layer4': 'layer4'
+    } if use_plus else {
+        'layer4': 'layer4'
+    }
+    backbone = _utils.IntermediateLayerGetter(backbone,
+                                              return_layers=return_layers)
+
+    try:
+        from mmseg.models import ASPPHead, DepthwiseSeparableASPPHead
+    except ImportError as e:
+        raise ImportError(
+            textwrap.dedent("""\
+            Either mmcv or mmsegmentation is not installed. To install mmcv, please run pip install mmcv-full==1.4.4 -f
+             https://download.openmmlab.com/mmcv/dist/{cu_version}/{torch_version}/index.html where {cu_version} and
+             {torch_version} refer to your CUDA and PyTorch versions, respectively. To install mmsegmentation, please
+             run pip install mmsegmentation==0.22.0 on command-line.""")
+        ) from e
+
+    world_size = dist.get_world_size()
+    if sync_bn and world_size == 1:
+        warnings.warn(
+            'sync_bn was true, but only one process is present for training. sync_bn will be ignored.'
+        )
+
+    norm_type = 'SyncBN' if sync_bn and world_size > 1 else 'BN'
+    norm_cfg = {'type': norm_type, 'requires_grad': True}
+    if use_plus:
+        # mmseg config:
+        # https://github.com/open-mmlab/mmsegmentation/blob/master/configs/_base_/models/deeplabv3plus_r50-d8.py
+        head = DepthwiseSeparableASPPHead(in_channels=2048,
+                                          in_index=-1,
+                                          channels=512,
+                                          dilations=(1, 12, 24, 36),
+                                          c1_in_channels=256,
+                                          c1_channels=48,
+                                          dropout_ratio=0.1,
+                                          num_classes=num_classes,
+                                          norm_cfg=norm_cfg,
+                                          align_corners=False)
+    else:
+        # mmseg config:
+        # https://github.com/open-mmlab/mmsegmentation/blob/master/configs/_base_/models/deeplabv3_r50-d8.py
+        head = ASPPHead(in_channels=2048,
+                        in_index=-1,
+                        channels=512,
+                        dilations=(1, 12, 24, 36),
+                        dropout_ratio=0.1,
+                        num_classes=num_classes,
+                        norm_cfg=norm_cfg,
+                        align_corners=False)
+
+    model = SimpleSegmentationModel(backbone, head)
+
+    # Only apply initialization to classifier head if pre-trained weights are used
+    if init_fn:
+        if backbone_weights is None:
+            model.apply(init_fn)
+        else:
+            model.classifier.apply(init_fn)
+
+    if sync_bn and world_size > 1:
+        local_world_size = dist.get_local_world_size()
```

Comment:
Do we need sync batchnorm? Just looking for ways to simplify what we give to users + customers.

Reply:
it seems that we aren't using it but that's very much batch size dependent. It's likely going to be useful if someone has a smaller gpu or large enough image size to force their per device batch size down to ~4 or less.